File: api/src/main/java/com/envolti/teste/entity/Pessoa.java
package com.envolti.teste.entity;

import com.envolti.teste.enums.Sexo;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.validator.constraints.br.CPF;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.Email;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Pessoa {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nome;

    @Enumerated(EnumType.STRING)
    private Sexo sexo;

    @Email(message = "Email should be valid")
    private String email;

    @Column(name = "data_nascimento", nullable = false)
    private LocalDate dataNascimento;

    private String naturalidade;

    private String nacionalidade;

    @Column(nullable = false, unique = true)
    @CPF(message = "Cpf inválido")
    private String cpf;

    @CreationTimestamp
    @Column(name = "create_date_time")
    private LocalDateTime createDateTime;

    @UpdateTimestamp
    @Column(name = "update_date_time")
    private LocalDateTime updateDateTime;

}

File: front/src/app/pessoa/pessoa.ts
export class Pessoa {

    id: number;
    nome: string;
    email: string;
    sexo: string;
    dataNascimento: string;
    naturalidade: string;
    cpf: string;
    nacionalidade: string;

  constructor(id: number, nome: string, email: string, sexo: string,
              dataNascimento: string, naturalidade: string, cpf: string, nacionalidade: string) {
    this.id = id;
    this.nome = nome;
    this.email = email;
    this.sexo = sexo;
    this.dataNascimento = dataNascimento;
    this.naturalidade = naturalidade;
    this.cpf = cpf;
    this.nacionalidade = nacionalidade;
  }
}
